lib/scion.js

```diff
@@ -685,14 +714,55 @@
 
         /** @private */
         _performBigStep : function(e) {
+            this.emit('onBigStepBegin');
             if (e) this._internalEventQueue.push(e);
             var keepGoing = true;
             while (keepGoing) {
                 var currentEvent = this._internalEventQueue.shift() || null;
+                this.emit('onSmallStepBegin', currentEvent);
                 var selectedTransitions = this._performSmallStep(currentEvent);
                 keepGoing = !selectedTransitions.isEmpty();
             }
             this._isInFinalState = this._configuration.iter().every(function(s){ return s.typeEnum === STATE_TYPES.FINAL; });
+            this.emit('onBigStepEnd');
+        },
+
+        _performBigStepAsync : function(event, cb) {
+            if (event) {
+                this._internalEventQueue.push(event);
+            }
+
+            var self = this;
+            function doBigStep(eventToEmit) {
+                var selectedTransitions;
+                try {
+                    self.emit(eventToEmit);
+                    var currentEvent = self._internalEventQueue.shift() || null;
+                    self.emit('onSmallStepBegin', currentEvent);
+                    selectedTransitions = self._performSmallStep(currentEvent);
+                    self.emit('onSmallStepEnd');
+                } catch(err) {
+                    cb(err);
+                    return;
```

Comment:
Because this function returns before the state machine has stabilized (`selectedTransitions.isEmpty()`), is it possible for an error to cause the big step to end while the state machine is in an illegal configuration?